src/components/organisms/Modal/Modal.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  FC,
+  ReactNode,
+  PropsWithChildren,
+  Dispatch,
+  SetStateAction,
+} from 'react'
+import * as Dialog from '@radix-ui/react-dialog'
+import classNames from 'classnames'
+import Button, { AppearanceTypes } from 'components/molecules/Button/Button'
+import { ReactComponent as Close } from 'assets/icons/close.svg'
+import { useTranslation } from 'react-i18next'
+
+import classes from './styles.module.scss'
+
+export enum ModalSizeTypes {
+  Narrow = 'narrow',
+  Medium = 'medium',
+  Big = 'big',
+}
+
+export enum ButtonPositionTypes {
+  SpaceBetween = 'spaceBetween',
+  Right = 'right',
+  None = 'none',
+}
+
+export enum TitleFontTypes {
+  Gray = 'gray',
+  Black = 'black',
+}
+
+export interface ModalProps {
+  title?: string
+  trigger?: ReactNode
+  customDialogContent?: string
+  topButton?: boolean
+  size?: ModalSizeTypes
+  buttonsPosition: ButtonPositionTypes
+  breakButtonLabel?: string
+  proceedButtonLabel?: string
+  handleClose: () => void
+  handleProceed: () => void
+  open: boolean
+  setOpen: Dispatch<SetStateAction<boolean>>
+  titleFont?: TitleFontTypes
+}
+
+export interface ModalFooterProps {
+  buttonsPosition: ButtonPositionTypes
+  breakButtonLabel?: string
+  proceedButtonLabel?: string
+  handleClose: () => void
+  handleProceed: () => void
+}
+
+const ModalFooter: FC<ModalFooterProps> = ({
+  buttonsPosition = 'right',
+  breakButtonLabel,
+  proceedButtonLabel,
+  handleClose,
+  handleProceed,
+}) => {
+  return (
+    <div className={classes[buttonsPosition]}>
+      <Button appearance={AppearanceTypes.Secondary} onClick={handleClose}>
```

Comment:
**[blocker]**

It seems that currently we only support modals with exactly 0 buttons or 2 buttons, where the left button is of Secondary type and the right button is primary.

According to design we have some cases that are not covered by this. 
For example: https://www.figma.com/file/VqCwED3Dymn7L1gGoAOK27/EKI-t%C3%B5lkev%C3%A4rav?type=design&node-id=2346-107207&t=je6Y5mYgtspPolA1-4 

here we have split buttons, where the button the left is a text button and the button on the right is secondary button.

**Extra note**:

Currently we also only support passing onClick action to these buttons however in the same example the left button is actually a link, so we want to pass "href" to it instead of onClick

Another example which is not supported is:

https://www.figma.com/file/VqCwED3Dymn7L1gGoAOK27/EKI-t%C3%B5lkev%C3%A4rav?type=design&node-id=936-91678&t=Z4cUh8GLdliLnTqX-4

Here 1 of the buttons will be disabled until some times are picked, which means that it needs some information from the content of the modal to decide whether it is disabled or not.



**Potential solution**

Based on this I would suggest keeping the wrapper for the buttons for their positioning inside this component, however allowing the passing of the Buttons as props. Basically just 1 prop, which is rendered inside this wrapper div. It can have as many buttons as we want (possibly event 1 or 3 for the future, or whatever number). This wrapper will just decide how they are positioned + how much spacing they have from edges and from top/bottom + makes sure that they will always be visible at the bottom of the modal